source/MaterialXGenShader/HwShaderGenerator.cpp

```diff
@@ -292,9 +307,16 @@ ShaderPtr HwShaderGenerator::createShader(const string& name, ElementPtr element
         // and are editable by users.
         if (!inputSocket->getConnections().empty() && graph->isEditable(*inputSocket))
         {
+            if (inputSocket->getType() == Type::FILENAME)
+            {
+                psSamplerUniforms->add(inputSocket->getSelf());
```

Comment:
All texture inputs end up in a single sampler block now, so we have no separation between public and private, which we have for all other types of inputs. I wonder if it would make sense to create two sampler blocks, separating private and public textures inputs?

Reply:
Or maybe it's better that we refactor the way we handle private vs. public inputs. For example we could simply introduce a flag on the ShaderPort class that tells the client if the input should be published in UI or kept private. That way we could remove the separation of private and public uniform blocks. I think that's better, and I can do this in a separate change list after this one. So keep this as is here :)

Reply:
Yes, doing it in ShaderPort is better way but there might be other challenges with Visual Editors that use ShaderPort direclty. I'm not sure.

Reply:
We can discuss this on slack.